### PR TITLE
Adopting support to set the source for a diagnostic

### PR DESCRIPTION
**Pull request by egamma**
Notice this requires typescript@2.3.1-insiders.20170420.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -102,6 +102,7 @@ function init(modules: { typescript: typeof ts_module }) {
             length: problem.getEndPosition().getPosition() - problem.getStartPosition().getPosition(),
             messageText: message,
             category: category,
+            source: 'tslint',
```
Comment (angelozerr) [inline src/index.ts:105]: I tell me if it's a good idea to add tslint version?

File: src/index.ts [hunk 2]
```diff
@@ -258,9 +259,9 @@ function init(modules: { typescript: typeof ts_module }) {
                   description: `Disable rule '${problem.getRuleName()}'`,
                   changes: [{
                     fileName: fileName,
-                    textChanges: [{
```
Comment (angelozerr) [inline src/index.ts:261]: Why is there changes here?

File: src/index.ts (hunk 1 above)
Comment (egamma) [reply src/index.ts:105]: Since all the diagnostics will all come from the same version the base `tslint` is sufficient.

File: src/index.ts (hunk 2 above)
Comment (egamma) [reply src/index.ts:261]: hmm, I didn't intend to edit these lines at all and I cannot spot any edits. Could be a change the line terminator. I would have to redo the PR to fix this. Let me know if you want me to do this.

File: src/index.ts (hunk 1 above)
Comment (angelozerr) [reply src/index.ts:105]: ok I wanted just to know your opinion about that.

File: src/index.ts (hunk 2 above)
Comment (angelozerr) [reply src/index.ts:261]: No it's not a problem, I can accept your PR like this but the big pb is about compilation https://travis-ci.org/angelozerr/tslint-language-service/builds/225117180#L273 

Do you know how to fix this error?

Comment (egamma) [reply src/index.ts:261]: I think you can fix this by changing the dependency to typescript in the package.json to typescript@2.3.1-insiders.20170420.  Only this version has the new definition of the Diagnostics type.

Comment (egamma) [reply src/index.ts:261]: I've updated the PR with a change to the package.json

**Comment by angelozerr** [comment]
Thanks @egamma !

**Comment by angelozerr** [comment]
@egamma I have published 0.9.1 which embed your feature. I have implemented 'source' inside typescript.java and it works well! Thanks! 

But it seems that TypeScript 2.3.1 doesn't fill 'source' with 'ts' for diagnostic. Is it normal?

**Comment by egamma** [comment]
Thanks!

> But it seems that TypeScript 2.3.1 doesn't fill 'source' with 'ts' for diagnostic. Is it normal?

yes it is 'normal', we do it this in VS Code on the client side, but it would be better when the tsserver sets the source field, then it is consistent across the clients.

**Comment by angelozerr** [comment]
Thnaks @egamma for your answer. Could you fill an issue about this problem in TypeScript issues?

**Pull request merged**